src/hooks/useSlide.js

```diff
@@ -1,44 +1,113 @@
 import React from 'react';
+import { DeckContext } from './useDeck';
 
 /**
- *
+ * Performs logic operations for all of the slide domain level.
+ * slideElements are dynamic elements within the slide that may
+ * appear/ move around etc.
+ * If remaining elements in slide, these are brought in one by one.
+ * If not, we tell the deck to take us to the next slide.
  */
 
+//  Keeps previous slide number for comparison later.
+let prevSlideNum;
```

Comment:
I think using `usePrevious` might be more idiomatic. https://usehooks.com/usePrevious/
We might also run into problems with this global variable if we have multiple invocations of `useSlide`.